### PR TITLE
Fix class variable caching in tests

### PR DESCRIPTION
**Pull request by oscarwyatt**
Our work on [govuk-docker](https://github.com/alphagov/govuk-docker/pull/100) found some problems where running tests on search-api with two clusters of elasticsearch weren't passing, this fixes that

### CODE REVIEW

File: lib/search_config.rb [hunk 1]
```diff
@@ -71,7 +67,9 @@ def run_batch_search(searches)
     end
 
     def elasticsearch
-      @elasticsearch ||= ElasticsearchConfig.new.config
+      Cache.get(Cache::SEARCH_CONFIG) do
+        @elasticsearch = ElasticsearchConfig.new.config
```
Comment (barrucadu) [inline lib/search_config.rb]: Does that `@elasticsearch =` want removing?

**Review by barrucadu — COMMENTED** [review]
Approach looks good to me, my only concern is that we should be sure we've not lost caching of anything which was previously cached.

**Comment by barrucadu** [comment]
~A thing for me to do: get the cluster B URI environment variable set in CI so we can actually test this.~

Done in this PR because on reflection breaking builds of every search-api branch seems like a bad idea.

**Comment by koetsier** [comment]
Closing this because I opened a new PR https://github.com/alphagov/search-api/pull/1619 which is the same but with a cleaned history

**Pull request closed without merging**